src/mon/OSDMonitor.cc

```diff
@@ -10207,8 +10218,10 @@ bool OSDMonitor::prepare_command_impl(MonOpRequestRef op,
 	goto reply;
     }
 
-    int64_t expected_num_objects;
-    cmd_getval(cct, cmdmap, "expected_num_objects", expected_num_objects, int64_t(0));
+    if (expected_num_objects == 0) {
```

Comment:
do you think it would be better to move this block into the erasure branch, i.e. right after https://github.com/ceph/ceph/pull/19651/files#diff-0a5db46a44ae9900e226289a810f10e8L10193 ? it's more symmetric this way, IMHO.

Reply:
@tchaikov okay